lib/classes/Service.js

```diff
@@ -1,304 +1,130 @@
-		this.resources[resources.getName()] = resources;
-	}
-
-	getAllResources(resourcesName) {
-		if (this.resources[resourcesName]) return this.resources[resourcesName];
-		else return this.resources[Object.keys(this.resources)[0]]; // This temporarily defaults to a single resource stack for backward compatibility
-	}
-
-	getAllStages() {
-		let stages = [];
-		for (let i = 0; i < Object.keys(this.stages).length; i++) {
-			stages.push(this.stages[Object.keys(this.stages)[i]]);
-		}
-		return stages;
-	}
-
-	getStage(name) {
-		return this.stages[name];
-	}
-
-	setStage(stage) {
-		this.stages[stage.getName()] = stage;
-	}
-
-	removeStage(name) {
-		let stage = this.stages[name];
-		delete this.stages[name];
-	}
-
-	validateStageExists(name) {
-		return this.stages[name] != undefined;
-	}
-
-	getRegion(stageName, regionName) {
-		if (this.getStage(stageName)) {
-			let stage = this.getStage(stageName);
-			if (stage.hasRegion(regionName)) {
-				return stage.getRegion(regionName);
-			} else {
-				throw new SError(`Region ${regionName} doesnt exist in stage ${stageName}!`);
-			}
-		} else {
-			throw new SError(`Stage ${stageName} doesnt exist in this project!`);
-		}
-	}
-
-	getAllRegions(stageName) {
-		return this.getStage(stageName).getAllRegions();
-	}
-
-	getAllRegionNames(stageName) {
-		return _.map(this.getAllRegions(stageName), 'name');
-	}
-
-	setRegion(stageName, region) {
-		let stage = this.getStage(stageName);
-		stage.setRegion(region);
-	}
-
-	validateRegionExists(stageName, regionName) {
-		let stage = this.getStage(stageName);
-
-		if (stage) {
-			return stage.hasRegion(regionName);
-		} else {
-			return false;
-		}
-	}
-
-	setVariables(variables) {
-		this.variables = variables;
-	}
-
-	getVariables() {
-		return this.variables;
-	}
-
-	getVariablesObject(stage, region) {
-		let vars = this.getVariables().toObject();
-		if (stage) stage = this.getStage(stage);
-		if (stage && region) region = stage.getRegion(region);
-		vars = _.merge(vars, stage ? stage.getVariables().toObject() : {}, region ? region.getVariables().toObject() : {});
-		return vars;
-	}
-
-	addVariables(variablesObj) {
-		return this.getVariables().fromObject(variablesObj);
-	}
-
-	setTemplates(templates) {
-		this.templates = templates;
-	}
-
-	getTemplates() {
-		return this.templates;
-	}
-
-	addPlugin(pluginName) {
-		this.plugins.push(pluginName);
-	}
+    if (data) this.fromObject(data);
+  }
 
-}
+  load() {
+    let _this = this;
+
+    const servicePath = _this.S.instances.config.servicePath;
+
+    if (!servicePath) throw SError('ServicePath is not configured.');
+
+    return _this.S.instances.yamlParser
+      .parse(path.join(servicePath, 'serverless.yaml'))
+      .then((serverlessYaml) => {
+        _this.service = serverlessYaml.service;
+        _this.custom = serverlessYaml.custom;
+        _this.plugins = serverlessYaml.plugins;
+
+        // load any defined resources based on provider
+        if (serverlessYaml.resources.aws) {
```

Comment:
What if we load the resources hash directly from the serverless.yml file instead of setting it explicitly here? I want to make sure that this part is not dependent on any specific provider and can be easily extended.

So adding a Docker provider for example shouldn't need going in here and adding Docker